tests/varargs-test.cpp

```diff
@@ -1,10 +1,38 @@
 #include <gtest/gtest.h>
 
-extern "C" void varargs_test();
+extern "C" void test_va_list(char *buf, size_t length);
+extern "C" void test_sprintf(char *buf, size_t length);
+extern "C" void test_va_arg_pack(char *buf, size_t length);
 
-TEST(Aarch64MinGW, VarargsTest) 
+// Test if the va_list builtin can be passed to CRT methods. 
+// I would expect this to fail until the va_list builtin is implemented for 
+// the Windows ABI.
+TEST(Aarch64MinGW, TestVaList) 
 {
-    varargs_test();
+    char sz[100];
+    test_va_list(sz, 100);
+    ASSERT_STREQ(sz, "string 11 1919 111.111000");
```

Comment:
to have all data for testing inside of C implementation will be preferable

Reply:
TestVaList FAILED
TestSPrintf OK
TestVaArgPack OK

I would expect one more test to fail based on our discussion

Reply:
As written in comments, sprintf test only fails for UCRT.

// Test if sprintf can receive variadic parameters. 
// In most cases this is inlined and should work. 
// In the case of UCRT it is a direct call to the CRT
// and I would expect it to fail until the variadic 
// ABI is correctly implemented.
TEST(Aarch64MinGW, TestSPrintf)

Reply:
based on our discussion regarding variadic functions, I remember there should be 2 tests failing, what about another one?